core/src/util/usUtils.cpp

```diff
@@ -388,6 +388,37 @@ std::string GetLastErrorStr()
 #endif
 }
 
+//-------------------------------------------------------------------
+// Android Compatibility functions
+//-------------------------------------------------------------------
+
+namespace {
+
+template <typename T>
+std::string internal_to_string(T val)
+{
+#if defined(__ANDROID__)
+#include <sstream>
```

Comment:
#include in the middle of a file is odd ... I understand you want to minimize the number of conditional compile blocks and unnecessary includes. What do you think about moving this function to the top of the file and using a separate function body for the if and the else block

```
#if defined(__ANDROID__)
#include <sstream>
Std::string ToString(...)
{
...
}
#else
Std::string ToString(...) { return std::to_string(...); }
#endif
```